### PR TITLE
2576 metadata should not be exposed in draft page source

### PR DESCRIPTION
**Pull request by lbownik**

### CODE REVIEW

File: dataverse-webapp/src/main/webapp/dataset.xhtml [hunk 1]
```diff
@@ -23,41 +24,49 @@
...
-            <script type="application/ld+json">
+            <c:if test="#{DatasetPage.workingVersion.released}">
+	            <script type="application/ld+json">
                 <h:outputText value="#{DatasetPage.jsonLd}"/>
-            </script>
+                </script>
+            </c:if>
         </ui:define>
         <ui:define name="robots_info">
+            <c:if test="#{DatasetPage.workingVersion.released}">
```
Comment (diasf) [inline dataverse-webapp/src/main/webapp/dataset.xhtml:50]: Nit: Could have been integrated into the `jsf:rendered` condition.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/page/FilePage.java [hunk 2]
```diff
@@ -352,20 +352,15 @@ public boolean isThumbnailAvailable(FileMetadata fileMetadata) {
         return thumbnailAvailable;
     }
 
-    public boolean isLicenseIconAvailable(FileTermsOfUse termsOfUse) {
-        return termsOfUse.getTermsOfUseType() == TermsOfUseType.LICENSE_BASED
```
Comment (diasf) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/page/FilePage.java:356]: I guess this condition was redundant, meaning other types of licenses don't have icons we shouldn't be showing?

**Review by diasf — APPROVED** [review]
LGTM, just some minor comments.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/page/FilePage.java (hunk 2 above)
Comment (lbownik) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/page/FilePage.java:356]: It was redundant as getTermsOfUseType() returns TermsOfUseType.LICENSE_BASED when termsOfUse.license is null which we check directly in 
public Optional<LicenseIcon> getIcon() {
        return Optional.ofNullable(this.license).map(License::getIcon);
    }

uning Optional.ofNullable

**Pull request merged**